docs/recording-tx.md

```diff
@@ -0,0 +1,57 @@
+# RecordingTransaction recipe
+
+A `RecordingTransaction` is an implementation of `Transaction` that logs all transaction operations
+(i.e GET, SET, or DELETE) to a `TxLog` object for later uses such as exporting data.  The code below
+shows how a RecordingTransaction is created by wrapping a Transaction object:
+
+```java
+RecordingTransactionBase rtx = RecordingTransactionBase.wrap(tx);
+```
+
+A predicate function can be passed to wrap method to select which log entries to record.  The code
+below only records log entries whose column family is `meta`:
+
+```java
+RecordingTransactionBase rtx = RecordingTransactionBase.wrap(tx,
+                               le -> le.getColumn().getFamily().toString().equals("meta"));
+```
+
+After creating a RecordingTransaction, users can use it as they would use a Transaction object.
+When they are done operating on the transaction, they can retrieve the TxLog using the following
+code:
+
+```java
+rtx.getTxLog()
+```
+
+Below is example code of how a RecordingTransaction can be used in an observer to record all operations
+performed by the transaction in TxLog.  This TxLog can be added to an export queue and later used to
+export updates from Fluo.
+
+```java
+public class MyObserver extends AbstractObserver {
+
+    private static final TYPEL = new TypeLayer(new StringEncoder());
+    
+    private ExportQueue<String, TxLog> exportQueue;
+
+    @Override
+    public void process(TransactionBase tx, Bytes row, Column col) {
+
+        // create recording transaction (rtx)
+        RecordingTransactionBase rtx = RecordingTransactionBase.wrap(tx);
+        
+        // use rtx to create a typed transaction & perform operations
+        TypedTransactionBase ttx = TYPEL.wrap(tx);
+        ttx.mutate().row("row").fam("cf").qual("cq").set("myvalue");
```

Comment:
Could make this part of the example slightly more complex... and after the example mention what would be in txLog for a given row.   Also drop the quotes around row

``java
int count = ttx.get().row(row).fam("meta").qual("counter1").toInteger(0);
ttx.mutate().row(row).fam("meta").qual("counter1").set(count+1);
```